### PR TITLE
connect to MetaMask

### PR DESCRIPTION
**Pull request by syuhei176**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -20,11 +20,11 @@ class ClientWrapper {
    * initialize plasma light client with privateKey
    * @param {string} privateKey privateKey of user's account
```
Comment (tkmct) [inline src/client.js:21]: can you add doc comment here?

File: src/components/StartupModal.js [hunk 2]
```diff
@@ -22,24 +22,24 @@ const StartUpModal = props => {
       <div>
         <Input
           className="mbs"
-          full
+          full="true"
```
Comment (tkmct) [inline src/components/StartupModal.js:25]: you don't have to put string. `full` is boolean in Button component

File: src/components/StartupModal.js [hunk 3]
```diff
@@ -22,24 +22,24 @@ const StartUpModal = props => {
...
           Create Wallet
         </Button>
-        {/* <div className="spacer">
+        <div className="spacer">
           <div className="spacer__txt">or</div>
         </div>
-        <Button full onClick={() => props.initializeMetamaskWallet()}>
-          Create Wallet with MetaMask
-        </Button> */}
+        <Button full="true" onClick={() => props.initializeMetamaskWallet()}>
```
Comment (tkmct) [inline src/components/StartupModal.js:40]: you don't have to wrap with arrow function for onClick if you don't pass any arguments. you can just pass a function

File: src/initialize.js [hunk 4]
```diff
@@ -27,27 +28,44 @@ function getProvider(network) {
...
+      new ethers.Wallet(privateKey, getProvider(networkName))
     )
     signer = wallet.getEthersWallet()
-  } else {
+  } else if (kind === WALLET_KIND.WALLET_METAMASK) {
+    const provider = new ethers.providers.Web3Provider(web3.currentProvider)
+    const address = await provider.getSigner().getAddress()
+    const network = await provider.ready
+    if (networkName !== 'local' && network.name !== networkName) {
+      throw new Error('Your wallet is connecting to different network.')
```
Comment (tkmct) [inline src/initialize.js:48]: what about passing expected network name and actual network name in error message?

File: src/initialize.js [hunk 5]
```diff
@@ -94,12 +112,14 @@ async function instantiate(privateKey) {
...
 
-export default async function initialize(privateKey) {
-  const lightClient = await instantiate(privateKey)
+export default async function initialize(kind, privateKey) {
+  const lightClient = await instantiate(kind, privateKey)
   await lightClient.start()
 
   // TODO: need more secure way to store private key.
-  localStorage.setItem('privateKey', privateKey)
+  if (privateKey !== undefined) {
```
Comment (tkmct) [inline src/initialize.js:120]: checking with `typeof privateKey === 'string'` might be better

**Comment by syuhei176** [comment]
@tkmct thank you for comments. I fixed all.

**Pull request merged**